docs/spec_draft.md

```diff
@@ -339,6 +349,168 @@ only exists to establish data dependencies from `result` to `inputs`.
+  * Each `operand`, corresponding to an execution instance in `G`,
+    is split into `split_count` number of blocks, described below using
+    Python-like syntax.
+    ```python
+    blocks = [
+        slice(
+          operand=operand,
+          start_indices=[s0, s1, ..., sR-1],
+            # where
+            #  - sk = 0 if k != split_dimension
+            #  - sk = j * dim(operand) // split_count, if k == split_dimension
+            #  - R = rank(operand)
+          limit_indices=[l0, l1, ..., lR-1],
+            # where
+            #   - lk = dim(operand, k) if k != split_dimension
+            #   - lk = dim(operand, split_dimension) // split_count, if k == split_dimension
+          strides=[t0, t1, ..., tR-1]
+            # where
+            #   - lk = 0 if k != split_dimension
+            #   - lk = dim(operand, split_dimension) // split_count, if k == split_dimension
+        ) for j in range(split_count)]
+    ```
+  * `blocks` are scattered to all the execution instances in `G` such that
+    `blocks[i]` is send to ith execution instance, for i $\in$ G.
+  * Each execution instance in `G` concatenates the received
+    blocks along the `concat_dimension` to produce `result` which is given by
+    `result = concatenate(received_blocks, concat_dimension)`,
+        where `received_blocks = { received blocks sorted based on the order of
+          execution instances in G }`.
+
+### Inputs
+
+| Name               | Type                                         |
+|--------------------|----------------------------------------------|
+| `operand`          | tensor of any supported type                 |
+| `split_dimension`  | constant of type `si64`                      |
+| `concat_dimension` | constant of type `si64`                      |
+| `split_count`      | constant of type `si64`                      |
+| `replica_groups`   | 2-dimensional tensor constant of type `si64` |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) `split_dimension` $\in$ [0, rank(`operand`)).
+  * (C2) `concat_dimension` $\in$ [0, rank(`operand`)).
+  * (C3) `split_count` $\gt$ 0.
+  * (C4) dim(`result`, `split_dimension`) % `split_count` $=$ 0.
+  * (C5) On `replica_groups`:
+    * All the subgroups in `replica_groups` have the same size.
+    * The replica ids contains all values in the range [0, N) where N is the
+      number of replicas in `replica_groups`.
+    * An empty `replica groups` is allowed only when `use_global_device_ids` is
+      `false`.
+    * `split_count` $=$ `size_subgroup`.
+  * (C6) dim(`result`, i) is given by:
+    * dim(`operand`, i) / `split_count`, i $=$ `split_dimension`.
+    * dim(`operand`, i) * `split_count`, i $=$ `concat_dimension`.
+    * dim(`operand`, i), otherwise.
+
+### Examples
+
+```mlir
+// %operand (at ei(0, 0)): [
+//                          [1, 2, 3, 4],
```

Comment:
Ah one more. The example type is `f32`, so the operand/result should also be in floating-points.